WordPress/Classes/Networking/MenusServiceRemote.h

```diff
@@ -0,0 +1,93 @@
+#import "ServiceRemoteREST.h"
+
+extern NSString * const MenusRemoteKeyID;
+extern NSString * const MenusRemoteKeyMenu;
+extern NSString * const MenusRemoteKeyMenus;
+extern NSString * const MenusRemoteKeyLocations;
+extern NSString * const MenusRemoteKeyContentID;
+extern NSString * const MenusRemoteKeyDescription;
+extern NSString * const MenusRemoteKeyLinkTarget;
+extern NSString * const MenusRemoteKeyLinkTitle;
+extern NSString * const MenusRemoteKeyName;
+extern NSString * const MenusRemoteKeyType;
+extern NSString * const MenusRemoteKeyTypeFamily;
+extern NSString * const MenusRemoteKeyTypeLabel;
+extern NSString * const MenusRemoteKeyURL;
+extern NSString * const MenusRemoteKeyItems;
+extern NSString * const MenusRemoteKeyDeleted;
+extern NSString * const MenusRemoteKeyLocationDefaultState;
+
+@class Blog;
+@class RemoteMenu;
+@class RemoteMenuItem;
+@class RemoteMenuLocation;
+
+typedef void(^MenusServiceRemoteSuccessBlock)();
+typedef void(^MenusServiceRemoteMenuRequestSuccessBlock)(RemoteMenu *menu);
+typedef void(^MenusServiceRemoteMenusRequestSuccessBlock)(NSArray<RemoteMenu *> *menus, NSArray<RemoteMenuLocation *> *locations);
+typedef void(^MenusServiceRemoteFailureBlock)(NSError *error);
+
+@interface MenusServiceRemote : ServiceRemoteREST
+
+#pragma mark - Remote queries: Creating and modifying menus
+
+/**
+ *  @brief      Create a new menu on a blog.
+ *
+ *  @param      menuName    The name of the new menu to be created.  Cannot be nil.
+ *  @param      blog        The blog to create the menu on.  Cannot be nil.
+ *  @param      success     The success handler.  Can be nil.
+ *  @param      failure     The failure handler.  Can be nil.
```

Comment:
Again, this is exactly how we've been doing things so far, but there's an opportunity here for improvement.

Basically the suggestion is instead of using comments, to describe nullability, you could use the new nullability keywords provided by Swift.

For example the method declaration below would become:

```objective-c
- (void)createMenuWithName:(NSString * _Nonnull)menuName
                      blog:(Blog * _Nonnull)blog
                   success:(MenusServiceRemoteMenuRequestSuccessBlock _Nullable)success
                   failure:(MenusServiceRemoteFailureBlock _Nullable)failure;
```